File: iosApp/iosApp/Services/MnemonicServiceAssembler.swift
// Created by web3d4v on 21/11/2022.
// Copyright (c) 2022 Sons Of Crypto.
// SPDX-License-Identifier: MIT

import Foundation
import web3walletcore

final class MnemonicServiceAssembler: AssemblerComponent {

    func register(to registry: AssemblerRegistry) {

        registry.register(scope: .singleton) { resolver -> MnemonicService in
            DefaultMnemonicService()
        }
    }
}

File: androidApp/src/main/java/com/sonsofcrypto/web3wallet/android/services/MnemonicServiceAssembler.kt
package com.sonsofcrypto.web3wallet.android.services

import com.sonsofcrypto.web3wallet.android.common.AssemblerComponent
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistry
import com.sonsofcrypto.web3wallet.android.common.AssemblerRegistryScope
import com.sonsofcrypto.web3walletcore.services.mnemonic.DefaultMnemonicService
import com.sonsofcrypto.web3walletcore.services.mnemonic.MnemonicService
import smartadapter.internal.extension.name

class MnemonicServiceAssembler: AssemblerComponent {

    override fun register(to: AssemblerRegistry) {

        to.register(MnemonicService::class.name, AssemblerRegistryScope.SINGLETON) {
            DefaultMnemonicService()
        }
    }
}